### PR TITLE
[FIX+IMP] stock_inventory_exclude_sublocation: Don't mess inheritance

### PR DESCRIPTION
**Pull request by pedrobaeza**
It also includes the removal of an unneeded dependency + better position of the field in view

@Tecnativa

### CODE REVIEW

**Comment by pedrobaeza** [comment]
@lreficent you were the one that made this module, isn't it?

File: stock_inventory_exclude_sublocation/views/stock_inventory_view.xml [hunk 1]
```diff
@@ -9,7 +10,7 @@
     <field name="model">stock.inventory</field>
     <field name="inherit_id" ref="stock.view_inventory_form"/>
     <field name="arch" type="xml">
-        <field name="filter" position="after">
+        <field name="location_id" position="after">
```
Comment (rousseldenis) [inline stock_inventory_exclude_sublocation/views/stock_inventory_view.xml:13]: Indeed, it's logical. To be complete, I would have put :
`groups="stock.group_stock_multi_locations"`

As location appear only in that case.

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Minor comment.

Indeed, that makes sense !

File: stock_inventory_exclude_sublocation/views/stock_inventory_view.xml (hunk 1 above)
Comment (pedrobaeza) [reply stock_inventory_exclude_sublocation/views/stock_inventory_view.xml:13]: Yeah, good idea. Done!

**Review by rousseldenis — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Tests are failing on `stock_move_location`

**Review by rafaelbn — APPROVED** [review]

**Comment by LoisRForgeFlow** [comment]
Updating qty on hand for a product using the wizard:

```
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/http.py", line 651, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/.repo_requirements/odoo/odoo/http.py", line 310, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/.repo_requirements/odoo/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/.repo_requirements/odoo/odoo/http.py", line 693, in dispatch
    result = self._call_function(**self.params)
  File "/.repo_requirements/odoo/odoo/http.py", line 342, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/http.py", line 335, in checked_call
    result = self.endpoint(*a, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 937, in __call__
    return self.method(*args, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 515, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/OCB-11.0/addons/web/controllers/main.py", line 938, in call_button
    action = self._call_kw(model, method, args, {})
  File "/home/odoo/OCB-11.0/addons/web/controllers/main.py", line 926, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/home/odoo/OCB-11.0/addons/stock/wizard/stock_change_product_qty.py", line 104, in change_product_qty
    inventory.action_done()
  File "/home/odoo/build/OCA/stock-logistics-warehouse/stock_cycle_count/models/stock_inventory.py", line 44, in action_done
    res = super(StockInventory, self).action_done()
  File "/home/odoo/build/OCA/stock-logistics-warehouse/stock_inventory_discrepancy/models/stock_inventory.py", line 70, in action_done
    return super(StockInventory, self).action_done()
  File "/home/odoo/OCB-11.0/addons/stock/models/stock_inventory.py", line 169, in action_done
    self.post_inventory()
  File "/home/odoo/OCB-11.0/addons/stock_account/models/stock.py", line 30, in post_inventory
    res = super(StockInventory, other_inventories).post_inventory()
  File "/home/odoo/OCB-11.0/addons/stock/models/stock_inventory.py", line 176, in post_inventory
    self.mapped('move_ids').filtered(lambda move: move.state != 'done')._action_done()
  File "/home/odoo/OCB-11.0/addons/sale_stock/models/stock.py", line 31, in _action_done
    for line in result.mapped('sale_line_id').sudo():
AttributeError: 'bool' object has no attribute 'mapped'
```

Just tried in a recent runbot of stock-logistics-warehouse and no error raised.

**Review by LoisRForgeFlow — CHANGES_REQUESTED** [review]
can you have a look to my above comment?

Other than that, the change makes complete sense.

**Comment by pedrobaeza** [comment]
Hehe, the problem is because other thing that I have discovered by the bad way, when my customer this morning isn't able to validate any picking, and I have fixed it directly with 77ef4c8d6e1f80c28b5165f6a4ba0862c641af8a. Please take care of not breaking inheritance on your modules.

I'm going now to merge this and please check also the red CI in your other module `stock_move_location`.

**Comment by LoisRForgeFlow** [comment]
@pedrobaeza I see :laughing: Can you rebase this PR to include the fix so I can validate it?

**Comment by LoisRForgeFlow** [comment]
oh, you already merged haha nevermind I guess

**Comment by LoisRForgeFlow** [comment]
> I'm going now to merge this and please check also the red CI in your other module stock_move_location.

that module is not mine...

**Comment by pedrobaeza** [comment]
OK, I thought it was. But isn't Eficent one?

**Comment by rousseldenis** [comment]
@pedrobaeza I'll ping the guy

**Comment by pedrobaeza** [comment]
Thanks :smiley:

**Comment by rousseldenis** [comment]
@sebalix Could you see why stock_move_location is going red ? I'll make an issue for that

**Pull request merged**